lib/Dialect/FIRRTL/Transforms/InferWidths.cpp

```diff
@@ -1775,19 +1811,29 @@ bool InferenceTypeUpdate::updateOperation(Operation *op) {
   // that is wider than the LHS, in which case an additional BitsPrimOp is
   // necessary to truncate the value.
   if (auto con = dyn_cast<ConnectOp>(op)) {
-    auto lhs = con.dest().getType().cast<FIRRTLType>();
-    auto rhs = con.src().getType().cast<FIRRTLType>();
-    auto lhsWidth = lhs.getBitWidthOrSentinel();
-    auto rhsWidth = rhs.getBitWidthOrSentinel();
+    auto lhs = con.dest();
+    auto rhs = con.src();
+    auto lhsType = lhs.getType().cast<FIRRTLType>();
+    auto rhsType = rhs.getType().cast<FIRRTLType>();
+
+    // If the source is an InvalidValue of unkown width, infer the type to be
+    // the same as the destination.
+    if (dyn_cast_or_null<InvalidValueOp>(rhs.getDefiningOp()) &&
+        hasUninferredWidth(rhs.getType()))
+      rhs.setType(lhsType);
```

Comment:
I love this change. Seems so much nicer than the hacky way `invalidvalue`s were treated before.